compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEHoistLogicalObjFifo.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree-amd-aie/IR/AMDAIEOps.h"
+#include "iree-amd-aie/Transforms/Passes.h"
+#include "iree-amd-aie/Transforms/Transforms.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+
+#define DEBUG_TYPE "iree-amdaie-hoist-logical-objectfifo"
+
+namespace mlir::iree_compiler::AMDAIE {
+
+/// Hoist logical objectFifo operations until one of the operands is located
+/// within the same scope.
+LogicalResult hoistLogicalObjFifoOp(RewriterBase &rewriter,
+                                    AMDAIE::LogicalObjectFifoFromMemrefOp op) {
+  Operation *parentOp = op;
+  while (parentOp) {
+    Operation *newParentOp = parentOp->getParentOp();
```

Comment:
I interpret this as
"If the new parent is a proper ancestor of any operand, keep hoisting"
which seems like not what you want?

Reply:
> "If the new parent is a proper ancestor of any operand, keep hoisting"

No, then it breaks,